roles/_install_age/tasks/main.yml

```diff
@@ -0,0 +1,62 @@
+---
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2022, Felix Fontein
+
+- name: Gather required information on localhost
+  when: sops_install_on_localhost
+  ansible.builtin.setup:
+    gather_subset:
+      - '!all'
+      - '!min'
+      - architecture
+      - distribution
+      - distribution_major_version
+      - distribution_version
+      - os_family
+  delegate_to: localhost
+  delegate_facts: true
+  run_once: true
+
+- vars:
+    _community_sops_install_age_facts: >-
+      {{ hostvars['localhost' if sops_install_on_localhost else inventory_hostname].ansible_facts }}
+  block:
+    - name: Show system information
+      ansible.builtin.debug:
+        msg: |-
+          Architecture: {{ _community_sops_install_age_facts.architecture }}
+          Distribution: {{ _community_sops_install_age_facts.distribution }} {{ _community_sops_install_age_facts.distribution_major_version }}
+          Distribution version: {{ _community_sops_install_age_facts.distribution_version }}
+          OS family: {{ _community_sops_install_age_facts.os_family }}
+
+    - name: Include distribution specific variables
+      ansible.builtin.include_vars: '{{ lookup("ansible.builtin.first_found", params) }}'
+      vars:
+        params:
+          files:
+            - >-
+              D-{{ _community_sops_install_age_facts.distribution }}-{{ _community_sops_install_age_facts.distribution_version }}.yml
+            - >-
+              D-{{ _community_sops_install_age_facts.distribution }}-{{ _community_sops_install_age_facts.distribution_major_version }}.yml
+            - >-
+              D-{{ _community_sops_install_age_facts.distribution }}.yml
+            - >-
+              OS-{{ _community_sops_install_age_facts.os_family }}-{{ _community_sops_install_age_facts.distribution_major_version }}.yml
+            - >-
+              OS-{{ _community_sops_install_age_facts.os_family }}.yml
+            - default.yml
+          paths:
+            - '{{ role_path }}/vars'
+
+    - name: Install system packages
+      ansible.builtin.package:
+        name: '{{ _community_sops_install_age_system_packages }}'
+      become: '{{ sops_become_on_install }}'
+      delegate_to: '{{ "localhost" if sops_install_on_localhost else omit }}'
+      run_once: '{{ sops_install_on_localhost or omit }}'
+      when: _community_sops_install_age_system_packages | length > 0
+
+    - name: Set results
+      ansible.builtin.set_fact:
+        age_installed: "{{ _community_sops_install_age_has_age }}"
```

Comment:
The role intention is to install age, wouldn't it be more appropriate to fail here?

Reply:
For a regular role that's definitely true. This one is internal though and right now only used in the integration tests, and there we don't want the tests to fail if age cannot be installed. If this is ever converted into a public role, this definitely has to change, but for now - since it is internal - I didn't want to bother adding an interface that we don't need yet and which will make the integration tests more complicated. (The best way to do this would probably be to have an option for this behavior, so you can have whatever you want - fail, warn, or ignore.)

Reply:
Alright, since it is internal and my knowledge about ansible tests is lacking let's wait until it's publically releasable.
But if you plan to add such behavioral option, I'd make it only for internal use to aid the testing - defaulting to fail, if i'm telling ansible to install a package (in this case via a role), as an user i'd excpect it to fail if it cannot.

Reply:
Yes, as a user I also expect that, and this is also one of the reasons why this is no longer part of the `install` role but a separate - and for now internal - role. When/if this is made public, it definitely needs a good concept (which there isn't right now).